bftengine/src/bcstatetransfer/SourceSelector.hpp

```diff
@@ -106,6 +121,9 @@ class SourceSelector {
   std::vector<uint16_t> actualSources_;
   bool receivedValidBlockFromSrc_;
 
+  uint16_t nominatedPrimary_ = NO_REPLICA;
+  uint16_t prePrepareMsgCounter_ = 0;
```

Comment:
SS is anaware to the fact we use the report by fetching prepreapre. in that sense we want to keep it internal to BCStateTRan. I think a more general name is a better match:
`nominatedPrimaryCounter_`

in addition, I would put these 3 parameters straight after currentPrimary_, they all belong to the same logic , so it make sense ot put them one near another.